syntax = "proto3";

package net;


message IPV4Addr {
  bytes addr = 1;
  uint32 port = 2;
};

message IPV6Addr {
  bytes addr = 1;
  uint32 port = 2;
};

message HolePunchParameters {
  oneof src_addr {
    IPV4Addr src_ipv4 = 1;
    IPV6Addr src_ipv6 = 2;
  };
  oneof dst_addr {
    IPV4Addr dst_ipv4 = 3;
    IPV6Addr dst_ipv6 = 4;
  };

  uint64 start_time_ms = 5;

  bool listen_first = 6;
  float connect_sec_start = 7;
  float connect_sec_max = 8;
  float connect_sec_scale = 9;
  uint32 connect_count = 10;
};

message ServerAddress {
    IPV4Addr ipv4 = 1;
    IPV6Addr ipv6 = 2;
};
